queue/smoke_test.go

```diff
@@ -219,6 +219,70 @@ func runOrderedSmokeTest(ctx context.Context, q amboy.Queue, size int, startBefo
 
 }
 
+func runWaitUntilSmokeTest(ctx context.Context, q amboy.Queue, size int, assert *assert.Assertions) {
+	if err := q.Start(ctx); !assert.NoError(err) {
+		return
+	}
+
+	testNames := []string{"test", "second", "workers", "forty-two", "true", "false", ""}
+	numJobs := size * len(testNames)
+
+	wg := &sync.WaitGroup{}
+
+	for i := 0; i < size; i++ {
+		wg.Add(1)
+		go func(num int) {
+			for _, name := range testNames {
+				cmd := fmt.Sprintf("echo %s.%d.a", name, num)
+				j := job.NewShellJob(cmd, "")
+				ti := j.TimeInfo()
+				assert.Zero(ti.WaitUntil)
+				assert.NoError(q.Put(j),
+					fmt.Sprintf("(a) with %d workers", num))
+				_, ok := q.Get(j.ID())
+				assert.True(ok)
+
+				cmd = fmt.Sprintf("echo %s.%d.b", name, num)
+				j = job.NewShellJob(cmd, "")
+				j.UpdateTimeInfo(amboy.JobTimeInfo{
+					WaitUntil: time.Now().Add(time.Hour),
```

Comment:
Is there a test for jobs with expired WaitUntils?

Reply:
No, but a zero time is in the past, relative to now, so it should be the same?